api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/MeasurementQueryBuilder.java

```diff
@@ -0,0 +1,163 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.model.Attribute;
+import org.pmiops.workbench.model.Operator;
+import org.pmiops.workbench.model.SearchParameter;
+import org.pmiops.workbench.utils.OperatorUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Service
+public class MeasurementQueryBuilder extends AbstractQueryBuilder {
+
+  public static final String NUMERICAL = "NUM";
+  public static final String CATEGORICAL = "CAT";
+  public static final String TEXT = "TEXT";
+  public static final String BOTH = "BOTH";
+  private static final String LAB = "LAB";
+
+  private static final String UNION_ALL = " union all\n";
+  private static final String AND = " and ";
+  private static final String OR = " or ";
+  private static final String MEASUREMENT_SQL_TEMPLATE =
+    "select person_id, measurement_date as entry_date\n" +
+      "from `${projectId}.${dataSetId}.measurement`\n" +
+      "where measurement_concept_id = ${conceptId}\n";
+  private static final String VALUE_AS_NUMBER =
+    "value_as_number ${operator} ${value}\n";
+  private static final String VALUE_AS_CONCEPT_ID =
+    "value_as_concept_id ${operator} unnest(${values})\n";
+
+  @Override
+  public QueryJobConfiguration buildQueryJobConfig(QueryParameters parameters) {
+    List<String> queryParts = new ArrayList<String>();
+    Map<String, QueryParameterValue> queryParams = new HashMap<>();
+    for (SearchParameter parameter : parameters.getParameters()) {
+      validateAttributes(parameter);
+      if (parameter.getConceptId() == null) {
+        throw new BadRequestException("Please provide valid concept id for Measurements.");
+      }
+      String baseSql = MEASUREMENT_SQL_TEMPLATE.replace("${conceptId}", parameter.getConceptId().toString());
+      List<String> tempQueryParts = new ArrayList<String>();
+      for (Attribute attribute : parameter.getAttributes()) {
+        if (attribute.getOperator().equals(Operator.ANY)) {
```

Comment:
So a couple things about the ANY operator:

1) We have logic that handles different Operator options in here:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/utils/OperatorUtils.java#L18

It doesn't handle ANY. That could lead to an NPE in here, for example:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java#L339

We face a similar problem with BETWEEN. We have hacked around this problem by stripping out the BETWEEN operator from our pyclient library here:

https://github.com/all-of-us/pyclient/blob/master/libproject/devstart.rb#L51

So perhaps we should do the same for ANY. But at some point we may need to reconsider this approach... perhaps we need different Operator enums for the client API vs. this API or something.

2) I was wondering if it's possible to express ANY in some other way; like "NOT_EQUALS" with a value of null, and have the SQL generation code smart enough to do "IS NOT NULL". That's what we're doing right now here:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java#L334

Or just not having an Operator or value set at all.

Reply:
Done.